sys/include/net/dhcpv6/client.h

```diff
@@ -59,6 +59,27 @@ extern "C" {
 #define CONFIG_DHCPV6_CLIENT_PFX_LEASE_MAX (1U)
 #endif
 
+/**
+ * @brief   Maximum number of address leases to be stored
+ */
+#ifndef CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX
+#define CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX (1U)
+#endif
+
+/**
+ * @brief  Number of addresses needed for using DHCPv6 IA_NA.
+ *
+ * @note    Used for calculation of @ref CONFIG_GNRC_NETIF_IPV6_ADDRS_NUMOF.
+ *          Set to 0 if `dhcpv6_client_ia_na` is not included.
+ */
+#ifndef CONFIG_DHCPV6_CLIENT_ADDRS_NUMOF
+#if defined(MODULE_DHCPV6_CLIENT_IA_NA) || defined(DOXYGEN)
+#define CONFIG_DHCPV6_CLIENT_ADDRS_NUMOF (1)
+#else
+#define CONFIG_DHCPV6_CLIENT_ADDRS_NUMOF (0)
+#endif
+#endif
```

Comment:
And remove both the configurability and the `CONFIG_` prefix from this and make it dependent on `CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX`:

```suggestion
#if defined(MODULE_DHCPV6_CLIENT_IA_NA) || defined(DOXYGEN)
#define DHCPV6_CLIENT_ADDRS_NUMOF ((int)(CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX))
#else
#define DHCPV6_CLIENT_ADDRS_NUMOF (0)
#endif
```

Reply:
Oh, very good. I wasn't sure if you can cast values when using #define. I will apply your changes and then squash.